distribution/lib/Standard/Base/0.0.0-dev/src/System/File.enso

```diff
@@ -32,9 +32,12 @@ polyglot java import java.nio.file.Path
 
          example_new = File.new Examples.csv_path
 new : (Text | File) -> File
-new path = case path of
-    Text -> File (here.get_file path)
-    _ -> path
+new path =
+  file = case path of
+    Text -> here.get_file path
+    File -> path
+    _    -> Error.throw (Illegal_Argument_Error "new file should be either a File or a Text")
+  file
```

Comment:
Any reason for introducing the temporary variable here?

Reply:
Nope, removed